packages/viewer/src/lib/volume-intersection/controller.ts

```diff
@@ -0,0 +1,79 @@
+import { Point } from '@vertexvis/geometry';
+
+import { SceneItemOperationsBuilder } from '../scenes';
+import { VolumeIntersectionQueryModel } from './model';
+
+export const DEFAULT_CONCURRENT_VOLUME_QUERY_LIMIT = 5;
+
+export type OperationTransform = (
+  builder: SceneItemOperationsBuilder
+) => SceneItemOperationsBuilder;
+
+export class VolumeIntersectionQueryController {
+  private previousViewerCameraControls?: boolean;
+  private operationTransform: OperationTransform;
+  private inFlightOperations = 0;
+
+  public constructor(
+    private model: VolumeIntersectionQueryModel,
+    private viewer: HTMLVertexViewerElement
+  ) {
+    this.operationTransform = (builder) => builder.select();
+  }
+
+  public setStartPoint(point: Point.Point): void {
+    this.previousViewerCameraControls = this.viewer.cameraControls;
+    this.viewer.cameraControls = false;
+
+    this.model.setStartPoint(point);
+  }
+
+  public setEndPoint(point: Point.Point): void {
+    this.model.setEndPoint(point);
+  }
+
+  public setOperationTransform(
+    operationTransform: (
+      builder: SceneItemOperationsBuilder
+    ) => SceneItemOperationsBuilder
+  ): void {
+    this.operationTransform = operationTransform;
+  }
+
+  public async execute(): Promise<void> {
+    const screenBounds = this.model.getScreenBounds();
+    const type = this.model.getType();
+    this.viewer.cameraControls = this.previousViewerCameraControls ?? true;
+    this.model.complete();
+
+    if (
+      screenBounds != null &&
+      this.inFlightOperations < DEFAULT_CONCURRENT_VOLUME_QUERY_LIMIT
```

Comment:
Should this be configurable in the controller? From my perspective this is something we could enforce on the backend rather than on the client. That might be a harder problem, so take this comment with a grain of salt. 😄

Reply:
Agreed, I think this is effectively a placeholder right now with the intent of revisiting in https://vertexvis.atlassian.net/browse/PLAT-2169. I wanted to add some sort of limitation here for internal testing, but I think this is something that's likely removed after that ticket. I think there are some other ideas around cancelling in-flight queries that we could try as well once we get a better idea of whether that's needed